test/unittests/execute_floating_point_test.cpp

```diff
@@ -155,6 +161,62 @@ class execute_floating_point_types : public testing::Test
         FP<T>::nan(1),
     };
 
+    // The [int_only_begin; int_only_end) is the range of floating-point numbers, where each
+    // representable number is an integer and there are no fractional numbers between them.
+    // These numbers are represented as mantissa [0x1.00..0; 0x1.ff..f]
+    // and exponent 2**(mantissa digits without implicit leading 1).
+    // The main point of using int_only_begin is to tests nearest() as for int_only_begin - 0.5 and
+    // int_only_begin - 1.5 we have equal distance to nearby integer values.
+    // (Integer shift is used instead of std::pow() because of the clang compiler bug).
```

Comment:
What clang compiler bug is this? I guess we have it linked in the hackmd, do we want to link to it here too?

Reply:
I only have the `clang_bug` branch for now.